src/compiler/checker.ts

```diff
@@ -167,10 +167,10 @@ namespace ts {
 
         return checker;
 
-        function getEmitResolver(sourceFile?: SourceFile) {
+        function getEmitResolver(sourceFile: SourceFile, cancellationToken: CancellationToken) {

```

Comment:
When you talk about the host pattern, you mean the one that we use in services? I think those operations are different because they are pure query operations, so they are always safe to interrupt, right? Interrupting the type checker could cause corruption because the checker does caching.